### ISSUE TITLE
Could not load type 'SimpleAuthentication.Core.ICache' 

### ISSUE THREAD

**Issue by hlesesne** [SimpleAuthentication/SimpleAuthentication#175]
I am getting the following error when trying to use the `Nancy.SimpleAuthentication` nuget package.

`Could not load type 'SimpleAuthentication.Core.ICache' from assembly 'SimpleAuthentication.Core, Version=0.2.0.0, Culture=neutral, PublicKeyToken=null'.`

I believe I may be able to fix the problem by downloading the source and targeting 4.5 when building, then manually adding a reference to the dll, but it makes it difficult as all other dependencies on my application are nuget only.

Are there any plans to refresh the binaries included in the Nuget package?

Thanks for your time and help with this.

Best regards,

Hal

TRACE:

```
[TypeLoadException: Could not load type 'SimpleAuthentication.Core.ICache' from assembly 'SimpleAuthentication.Core, Version=0.2.0.0, Culture=neutral, PublicKeyToken=null'.]
   System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes) +0
   System.Reflection.RuntimeAssembly.GetExportedTypes() +21
   Nancy.Extensions.AssemblyExtensions.SafeGetExportedTypes(Assembly assembly) +28
   Nancy.Bootstrapper.AppDomainAssemblyTypeScanner.<UpdateTypes>b__16(Assembly assembly) +25
   System.Linq.<SelectManyIterator>d__4`3.MoveNext() +176
   System.Linq.WhereSelectEnumerableIterator`2.MoveNext() +190
   System.Linq.Buffer`1..ctor(IEnumerable`1 source) +199
   System.Linq.Enumerable.ToArray(IEnumerable`1 source) +77
   Nancy.Bootstrapper.AppDomainAssemblyTypeScanner.UpdateTypes() +334
   Nancy.Bootstrapper.AppDomainAssemblyTypeScanner.LoadAssembliesWithNancyReferences() +610
   Nancy.Bootstrapper.AppDomainAssemblyTypeScanner..cctor() +296

[TypeInitializationException: The type initializer for 'Nancy.Bootstrapper.AppDomainAssemblyTypeScanner' threw an exception.]
   Nancy.Bootstrapper.AppDomainAssemblyTypeScanner.TypesOf(ScanMode mode) +0
   Nancy.Bootstrapper.NancyBootstrapperLocator.GetBootstrapperType() +27
   Nancy.Bootstrapper.NancyBootstrapperLocator.LocateBootstrapper() +24
   Nancy.Bootstrapper.NancyBootstrapperLocator.get_Bootstrapper() +16
   Nancy.Hosting.Aspnet.NancyHttpRequestHandler.GetBootstrapper() +35
   Nancy.Hosting.Aspnet.NancyHttpRequestHandler..cctor() +14

[TypeInitializationException: The type initializer for 'Nancy.Hosting.Aspnet.NancyHttpRequestHandler' threw an exception.]
   Nancy.Hosting.Aspnet.NancyHttpRequestHandler..ctor() +0

[TargetInvocationException: Exception has been thrown by the target of an invocation.]
   System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck) +0
   System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark) +119
   System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark) +232
   System.Activator.CreateInstance(Type type, Boolean nonPublic) +83
   System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark) +1117
   System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes) +124
   System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture) +20
   System.Web.HttpRuntime.CreateNonPublicInstance(Type type, Object[] args) +60
   System.Web.Configuration.HandlerFactoryCache..ctor(String type) +46
   System.Web.HttpApplication.GetFactory(String type) +86
   System.Web.MaterializeHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +224
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +155
```

**Comment by hlesesne** [comment]
I think I fixed it by updating SimpleAuthentication.Core to the latest.

Nancy.SimpleAuthentication appears to install a down version. The upgrade, clean and build, seems to fix things.

**Comment by PureKrome** [comment]
what version(s) are you currently using / have nuget-downloaded-installed?

**Comment by fagnercarvalho** [comment]
I had the same problem. Basically I created a empty project on VS 2015, did a Install-Package Nancy, Nancy.Hosting.Self and Nancy.SimpleAuthtication. My packages.config was like the following just after I did this:
``` xml
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Nancy" version="1.3.0" targetFramework="net46" />
  <package id="Nancy.Hosting.Self" version="1.3.0" targetFramework="net46" />
  <package id="Nancy.SimpleAuthentication" version="0.3.14" targetFramework="net46" />
  <package id="RestSharp" version="104.1" targetFramework="net46" />
  <package id="SimpleAuthentication.Core" version="0.2.0" targetFramework="net46" />
</packages>
```
So then I did a upgrade on `SimpleAuthentication.Core`, my `packages.config` reflected the change (from the version 0.2.0 to the version 0.3.14) and everything is working fine now.

**Comment by PureKrome** [comment]
@fagnercarvalho ah! sweet. you have a repo. cheers, lets see what we can do...

big :clap: for the intel!

**Comment by PureKrome** [comment]
Fixed with 71a54472ce434f7c328d68e19a647a2e0aa9c3a0 .

**Comment by PureKrome** [comment]
@fagnercarvalho Heya mate, there's a new `pre-release` build on NuGet with this fix. If you're interesting in testing it out, please give it a run :)

nuget command: `Install-Package SimpleAuthentication.Core -Pre`

(notice that it's a `-pre` package).

(also, there's a new `Nancy.SA... -pre` package, too.
